### PR TITLE
Code to serialise app to disk

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

File: R/save-app.R [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  list(
+    globals = globals,
+    packages = pkgs
+  )
+}
+
+resource_paths_get <- function() {
+  resources <- getNamespace("shiny")$.globals$resources
+  vapply(resources, "[[", "directoryPath", FUN.VALUE = character(1))
+}
```
Comment (cpsievert) [inline R/save-app.R:58]: I think you could be using ` shiny::resourcePaths()` instead?

Comment (hadley) [reply R/save-app.R:58]: That appears to be subtly different to this code. Do you know what the difference means?

Comment (cpsievert) [reply R/save-app.R:58]: `.globals$resourcePaths` contains a `httpuv::staticPath()` object, but `shiny::resourcePaths()` is just returning the directory path information that you're looking for

**Comment by hadley** [comment]
Need to hook up to `ShinyDriver$new()`, but otherwise not document. Will experiment with in book, and see how it goes before publicising.

Also need to comment list of challenges — e.g. what happens if your app uses an non-exported function?

File: R/save-app.R (hunk 1 above)
Comment (hadley) [reply R/save-app.R:58]: Cool, thanks!

**Review by jcheng5 — APPROVED** [review]

**Comment by jcheng5** [comment]
Any idea about the CI error on Windows?

**Comment by hadley** [comment]
@jcheng5 yeah, that's #359

**Pull request merged**